### PR TITLE
debugEnabled defaults to true in Debug build

### PR DESCRIPTION
**Pull request by pretyman**
Hi,

I've been trying to add ALC892 to AppleALC, but for some reason I cannot load the kext at all with the following message: 

AppleALC: ioutil @ failed to get boot-args property

Couldn't enable the debug flags, because for some reason I don't yet understand, boot arguments are not found. I'm on 10.11.2. Sorry I don't know what other details to give you since I'm a newbie, but I believe it makes sense to have the debugging enabled by default in a Debug build.

Unfortunately I only know how to do it through a macro.

My boot arguments are: "nvda_drv=1 dart=0"

The logs with debugging enabled are:
```
3/8/16 7:49:46.000 PM kernel[0]: AppleALC: (DEBUG) init @ starting AppleALC
3/8/16 7:49:46.000 PM kernel[0]: AppleALC: ioutil @ failed to get boot-args property
3/8/16 7:49:46.000 PM kernel[0]: AppleALC: (DEBUG) init @ no boot arguments available
3/8/16 7:49:46.000 PM kernel[0]: AppleALC: (DEBUG) init @ no boot arguments or found a disabling argument, exiting
3/8/16 7:57:46.000 PM kernel[0]: AppleALC: (DEBUG) init @ starting AppleALC
3/8/16 7:57:46.000 PM kernel[0]: AppleALC: ioutil @ failed to get boot-args property
3/8/16 7:57:46.000 PM kernel[0]: AppleALC: (DEBUG) init @ no boot arguments available
3/8/16 7:57:46.000 PM kernel[0]: AppleALC: (DEBUG) init @ no boot arguments or found a disabling argument, exiting
3/8/16 7:57:50.000 PM kernel[0]: AppleALC: (DEBUG) init @ starting AppleALC
3/8/16 7:57:50.000 PM kernel[0]: AppleALC: ioutil @ failed to get boot-args property
3/8/16 7:57:50.000 PM kernel[0]: AppleALC: (DEBUG) init @ no boot arguments available
3/8/16 7:57:50.000 PM kernel[0]: AppleALC: (DEBUG) init @ no boot arguments or found a disabling argument, exiting
(...)
```

### CODE REVIEW

**Comment by pretyman** [comment]
Actually, I was stupid to talk about my issue in this pull request, I see there are many commits since I last tried.

Please disregard the issue and just evaluate if having debug enabled in a debug build is a good idea, maybe its not a good idea at all.

Извините пажолуйста

**Comment by vit9696** [comment]
Hi,

Np :), taking the opportunity to talk about debugEnabled and friends :)  
debugEnabled makes no effect in a Release build, and enforcing it on a Debug one is probably not right. 
But I guess it is to be discussed whether Release builds should print the debug information or not if `alc_dbg` is present. 
What is your opinion?

**Comment by pretyman** [comment]
Sorry, I sounded confusing. Let me try again: I had problems enabling the alc_dbg, for some reason it would fail to find the boot-args. I was putting them in Clover's boot arg, is this correct, or should have been in nvram? I can't get my nvram to save yet.

After the update I only get Kernel Panic and no logging. When I finish classes I guess I'll have to setup 2 machines debugging to debug this, is this how you debug it yourself?

**Comment by vit9696** [comment]
Hi,  
I see. alc_dbg was bugged in some previous revisions. At this moment -alcdbg should be used instead. Both Clover boot-args and nvram will do for argument passing.  
As for panics, I usually do not need to setup a kernel debugger, since logs are normally more than enough. I have fixed two kernel panics, so it is possible it stops crashing for you too.

**Comment by pretyman** [comment]
Ураааа!!! It works!! :D 

The latest version is working for 892
Thanks so much for this Kext!!
